### ISSUE TITLE
Template in ASP.NET Core results in wrong swagger

### ISSUE THREAD

**Issue by bakashinji** [RSuter/NSwag#1435]
This code, in different files

```
public class PagedResponse<T> : PagedResultBase
        where T : class
{

	public ICollection<T> Results { get; set; }


	public PagedResponse()
	{
		Results = new List<T>();
	}
}

public sealed class SingleResponse
{
	Guid SomeInfo;
}
	
[HttpPost]
public Task<PagedResponse<SingleResponse>> Api([FromBody] PagedRequest request)
{
	// return the Item ...
}
```
results in the following swagger
```
{
  "x-generator": "NSwag v11.16.1.0 (NJsonSchema v9.10.41.0 (Newtonsoft.Json v9.0.0.0))",
  "swagger": "2.0",
  "info": {
    "title": "My Title",
    "version": "1.0.0"
  },
  "consumes": [
    "application/json"
  ],
  "produces": [
    "application/json"
  ],
  "paths": {
    "/api/v0/my/api": {
      "post": {
        "tags": [
          "My"
        ],
        "operationId": "my_api",
        "parameters": [
          {
            "name": "request",
            "in": "body",
            "required": true,
            "schema": {
              "$ref": "#/definitions/PagedRequest"
            },
            "x-nullable": true
          }
        ],
        "responses": {
          "200": {
            "x-nullable": true,
            "description": "",
            "schema": {
              "$ref": "#/definitions/PagedResponseOfSingleResponse"
            }
          }
        }
      }
    }
  },
  "definitions": {
    "PagedRequest": {
      "type": "object",
      "additionalProperties": false,
      "required": [
        "page",
        "pageSize"
      ],
      "properties": {
        "id": {
          "type": "string",
          "format": "guid"
        },
        "page": {
          "type": "integer",
          "format": "int32"
        },
        "pageSize": {
          "type": "integer",
          "format": "int32"
        }
      }
    },
    "PagedResponseOfSingleResponse": {
      "type": "object",
      "additionalProperties": false,
      "properties": {
        "results": {
          "type": "array",
          "items": {
            "$ref": "#/definitions/SingleResponse"
          }
        }
      },
      "allOf": [
        {
          "$ref": "#/definitions/PagedResultBase"
        }
      ]
    },
    "SingleResponse": {
      "type": "object",
      "additionalProperties": false,
      "required": [
        "someInfo"
      ],
      "properties": {
        "someInfo": {
          "type": "string",
          "format": "guid"
        }
      }
    },
    "PagedResultBase": {
      "type": "object",
      "x-abstract": true,
      "additionalProperties": false,
      "required": [
        "currentPage",
        "pageSize"
      ],
      "properties": {
        "currentPage": {
          "type": "integer",
          "format": "int32"
        },
        "pageSize": {
          "type": "integer",
          "format": "int32"
        }
      }
    }
  }
}
```

The part of PagedResponseOfSingleResponse is wrong and results in an empty Model if parsed by the swagger parse (apart from objects from PagedResultBase) and should be:

```
"PagedResponseOfSingleResponse": {
	"type": "object",
	"allOf": [
		{
			"$ref": "#/definitions/PagedResultBase"
		},
		{
			"additionalProperties": false,
			"properties": {
				"results": {
					"type": "array",
					"items": {
						"$ref": "#/definitions/SingleResponse"
					}
				}
			}
		}
	]
}
```
which results in a correct model

**Comment by RSuter** [comment]
Is this the same issue? https://github.com/RSuter/NJsonSchema/issues/732

**Comment by bakashinji** [comment]
I think they are the same issue. Also the fix @fizmike suggested in his second post (moving all the "properties" into the "allOf" array) is exactly what gets it to work with swagger codegen and the swagger-parser

I didn't realize this was a problem with swagger2 standard compliancy

**Comment by RSuter** [comment]
> moving all the "properties" into the "allOf" array

I've never seen a statement that having the properties in the root and allOf is not allowed...

**Comment by bakashinji** [comment]
Actually the parser in https://github.com/swagger-api/swagger-parser creates a model which doesn't have the SingleResponse in the PagedResponseOfSingleResponse, which is why I asked there first

**Comment by RSuter** [comment]
Fixed in v11.20.0
